### PR TITLE
AWS plugin config: BSL Region not required when s3ForcePathStyle is false and BackupImages is false

### LINKED ISSUE #424: Bug: AWS plugin config: BSL Region not required when `s3ForcePathStyle` is `false`
### Contact Details

_No response_

### Describe bug

When AWS plugin has `s3ForcePathStyle` and `s3Url` defined, a region does not need to be defined inside BSL.

### What happened?

This velero CR should not have error status
```yaml
apiVersion: oadp.openshift.io/v1alpha1
kind: Velero
metadata:
  name: example-velero
  namespace: openshift-adp
spec:
  backupStorageLocations:
    - backupSyncPeriod: 2m0s
      config:
        profile: default
        s3ForcePathStyle: 'true'
        s3Url: 'http://s3.openshift-storage.svc/'
      credential:
        key: cloud
        name: cloud-credentials
      default: true
      objectStorage:
        bucket: migstorage
        prefix: velero
      provider: aws
  defaultVeleroPlugins:
    - openshift
    - aws
    - csi
  enableRestic: true
  veleroFeatureFlags:
    - EnableCSI
  volumeSnapshotLocations:
    - config:
        profile: default
        region: us-east-1
      provider: aws
status:
  conditions:
    - lastTransitionTime: '2021-10-29T18:03:27Z'
      message: >-
        prefix and region for AWS backupstoragelocation object storage cannot be
        empty. It is required for backing up images
      reason: Error
      status: 'False'
      type: Reconciled

```

### OADP Version

0.4.x (Beta)

### OpenShift Version

4.8

### Velero pod logs

_No response_

### Restic pod logs

_No response_

### Operator pod logs

_No response_

### New issue

- [X] This issue is new

### PR DESCRIPTION
**Pull request by kaovilai**
Closes #424 
Closes [OADP-153](https://issues.redhat.com/browse/OADP-153)

- fix reconciler not deleting restic deamonset when restic config is `nil`

### CODE REVIEW

File: controllers/bsl.go [hunk 1]
```diff
@@ -181,9 +181,12 @@ func (r *DPAReconciler) validateAWSBackupStorageLocation(bslSpec velerov1.Backup
...
 	}
 
-	if len(bslSpec.StorageType.ObjectStorage.Prefix) == 0 || len(bslSpec.Config[Region]) == 0 &&
-		(dpa.Spec.BackupImages == nil || *dpa.Spec.BackupImages) {
-		return fmt.Errorf("prefix and region for AWS backupstoragelocation object storage cannot be empty. It is required for backing up images")
+	if len(bslSpec.StorageType.ObjectStorage.Prefix) == 0 && dpa.BSLPrefixRequired() {
+		return fmt.Errorf("prefix for AWS backupstoragelocation object storage cannot be empty. It is required for backing up images")
+	}
+	// BSL region is required when s3ForcePathStyle is true
+	if len(bslSpec.Config[Region]) == 0 && bslSpec.Config[S3ForcePathStyle] == "true" {
```
Comment (dymurray) [inline controllers/bsl.go:188]: Do we need to be checking for `s3Url` here too?

**Comment by kaovilai** [comment]
Looks like registry currently use region. If backupImages is enabled, region is still required.

**Review by dymurray — APPROVED** [review]

**Pull request merged**